File: src/e-community/android/xml/app/src/main/java/at/fhooe/smartmeter/dto/MeterDataDto.kt
package at.fhooe.smartmeter.dto

data class MeterDataDto<T> (val id: Int,
                            val timestamp: String,
                            var activeEnergyPlus: T,
                            var activeEnergyMinus: T,
                            var reactiveEnergyPlus: T,
                            var reactiveEnergyMinus: T,
                            var activePowerPlus: T,
                            var activePowerMinus: T,
                            var reactivePowerPlus: T,
                            var reactivePowerMinus: T,)

File: src/local-energy-monitoring/backend/SmartMeterAPI/SmartMeterAPI/DTOs/MeterDataDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SmartMeterAPI.DTOs
{
    public class MeterDataDto<T>
    {
        public int Id { get; set; }
        public DateTime Timestamp { get; set; }
        public T ActiveEnergyPlus { get; set; }
        public T ActiveEnergyMinus { get; set; }
        public T ReactiveEnergyPlus { get; set; }
        public T ReactiveEnergyMinus { get; set; }
        public T ActivePowerPlus { get; set; }
        public T ActivePowerMinus { get; set; }
        public T ReactivePowerPlus { get; set; }
        public T ReactivePowerMinus { get; set; }
    }
}
